doc/rst/technotes/gpu.rst

```diff
@@ -136,39 +139,125 @@ GPU-Related Environment Variables
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 To enable GPU support set the environment variable ``CHPL_LOCALE_MODEL=gpu``
-before building Chapel.
+before building Chapel. Several other variables affect how Chapel generates
+code for and interacts with GPUs. These variables include:
+
+* ``CHPL_GPU`` --- may be set to ``nvidia``, ``amd', or ``cpu``. If unset, as
+  part of its build process, Chapel will attempt to automatically determine
+  what type of GPU you're trying to target. Changing this variable requires
+  rebuilding Chapel. For more information, see the `Vendor Portability`_
+  section.
+* ``CHPL_GPU_ARCH`` --- specifies GPU architecture to generate kernel code for.
+  If unset and targeting NVIDIA GPUs, will default to 'sm_60'. This may also
+  be set by passing the ``chpl`` compiler ``--gpu-arch=<architecture>``. For
+  more information, see the `Vendor Portability`_ section.
+* ``CHPL_CUDA_PATH`` --- specifies path to CUDA toolkit.  If unset, Chapel tries
+  to automatically determine this path based on the location of ``nvcc``. This
+  variable is unused if not targeting NVIDIA GPUs. For more information, see
+  the `Vendor Portability`_ section.
+* ``CHPL_ROCM_PATH`` --- specifies path to ROCm library. If unset, Chapel tries
+  to automatically determine this path based on the location of ``hipcc``. This
+  variable is unused if not targeting AMD GPUs. For more information, see the
+  `Vendor Portability`_ section.
+* ``CHPL_RT_NUM_GPU_PER_LOCALE`` --- if using ``CHPL_GPU=cpu``, sets how many
```

Comment:
It was my understanding that if not using `CHPL_GPU=cpu`, if a locale had N GPUs, and you set this value to K < N, only the first K GPUs would be used.

Reply:
> It was my understanding that if not using CHPL_GPU=cpu, if a locale had N GPUs, and you set this value to K < N, only the first K GPUs would be used.

You're right, How's this:

```
* ``CHPL_RT_NUM_GPUS_PER_LOCALE`` --- sets how many GPU sublocales to have per
  locale. If using ``CHPL_GPU=cpu``, may be set to any non negative value,
  otherwise it may be set to any value equal-to or lower than the number of
  GPUs available on each node.  If unset, defaults to 1 if ``CHPL_GPU=cpu``,
  otherwise defaults to the number of GPUs available on each node.
  For more information, see the `CPU as Device mode`_ section.
```

I suppose it's really "f using ``CHPL_GPU=cpu``, may be set to any non negative value that fits in a signed 32-bit integer" but I don't think that's worth spelling out.

Reply:
How about:
> If unset, defaults to the number of GPUs available on each node, except for `CHPL_GPU=cpu`, in which case it defaults to 1.